From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 13 Oct 2008 10:08:36 -0500
Subject: [fs] ext4: add missing aops
Message-id: 48F36474.4020804@redhat.com
O-Subject: [RHEL5.3 PATCH] ext4: add missing aops
Bugzilla: 466246
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Josef Bacik <jbacik@redhat.com>

This is for
[Bug 466246] Interactive installation fails with ext4dev root partition

Embarrassingly enough, I missed a few aops on the backport -
and missing ->sendfile caused the problem with the installer
(LOOP_CHANGE_FD fails without it, and anaconda uses this)

This was tested an found to resolve the issue.

Thanks,
-Eric

diff --git a/fs/ext4/file.c b/fs/ext4/file.c
index 72d932d..9f61568 100644
--- a/fs/ext4/file.c
+++ b/fs/ext4/file.c
@@ -143,6 +143,8 @@ struct file_operations ext4_file_operations = {
 	.write		= do_sync_write,
 	.aio_read	= generic_file_aio_read,
 	.aio_write	= ext4_file_write,
+	.readv		= generic_file_readv,
+	.writev		= generic_file_writev,
 	.unlocked_ioctl = ext4_ioctl,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl	= ext4_compat_ioctl,
@@ -151,6 +153,7 @@ struct file_operations ext4_file_operations = {
 	.open		= generic_file_open,
 	.release	= ext4_release_file,
 	.fsync		= ext4_sync_file,
+	.sendfile	= generic_file_sendfile,
 	.splice_read	= generic_file_splice_read,
 	.splice_write	= generic_file_splice_write,
 };
